src/seedu/addressbook/data/person/Name.java

```diff
@@ -6,59 +6,91 @@
-    public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces or alphabetic characters";
-    public static final String NAME_VALIDATION_REGEX = "[\\p{Alpha} ]+";
+	public static final String EXAMPLE = "John Doe";
+	public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces or alphabetic characters";
+	public static final String NAME_VALIDATION_REGEX = "[\\p{Alpha} ]+";
 
-    public final String fullName;
+	public final String fullName;
 
-    /**
-     * Validates given name.
-     *
-     * @throws IllegalValueException if given name string is invalid.
-     */
-    public Name(String name) throws IllegalValueException {
-        name = name.trim();
-        if (!isValidName(name)) {
-            throw new IllegalValueException(MESSAGE_NAME_CONSTRAINTS);
-        }
-        this.fullName = name;
-    }
+	/**
+	 * Validates given name.
+	 *
+	 * @throws IllegalValueException
+	 *             if given name string is invalid.
+	 */
+	public Name(String name) throws IllegalValueException {
+		name = name.trim();
+		if (!isValidName(name)) {
+			throw new IllegalValueException(MESSAGE_NAME_CONSTRAINTS);
+		}
+		this.fullName = name;
+	}
 
-    /**
-     * Returns true if a given string is a valid person name.
-     */
-    public static boolean isValidName(String test) {
-        return test.matches(NAME_VALIDATION_REGEX);
-    }
+	/**
+	 * Returns true if a given string is a valid person name.
+	 */
+	public static boolean isValidName(String test) {
+		return test.matches(NAME_VALIDATION_REGEX);
+	}
 
-    /**
-     * Retrieves a listing of every word in the name, in order.
-     */
-    public List<String> getWordsInName() {
-        return Arrays.asList(fullName.split("\\s+"));
-    }
+	/**
+	 * Retrieves a listing of every word in the name, in order.
+	 */
+	public List<String> getWordsInName() {
+		return Arrays.asList(fullName.split("\\s+"));
+	}
 
-    @Override
-    public String toString() {
-        return fullName;
-    }
+	@Override
+	public String toString() {
+		return fullName;
+	}
 
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Name // instanceof handles nulls
-                && this.fullName.equals(((Name) other).fullName)); // state check
-    }
+	@Override
+	public boolean equals(Object other) {
+		return other == this // short circuit if same object
+				|| (other instanceof Name // instanceof handles nulls
+						&& this.fullName.equals(((Name) other).fullName)); // state
+																			// check
+	}
 
-    @Override
-    public int hashCode() {
-        return fullName.hashCode();
-    }
+	@Override
+	public int hashCode() {
+		return fullName.hashCode();
+	}
 
+	/**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if they are in a different case.
+     */
```

Comment:
Nice to see you've included the header comments! Remember to update when you update the method though